packages/atomic/src/components/commerce/product-template-components/atomic-product-image/atomic-product-image.tsx

```diff
@@ -212,13 +233,18 @@ export class AtomicProductImage implements InitializableComponent<Bindings> {
         alt: image.alt,
       };
     });
+
     if (this.images.length === 0) {
       this.validateUrl(this.fallback);
+      const alt = this.isImageAltFieldValid()
+        ? (this.imageAlt as string)
```

Comment:
RIght now you're not dealing with the possibility that `this.imageAlt` might be an array of strings.

If that's the case, then in order for it to be considered "valid", I suppose `this.imageAlt` would have to have the same length as the image field (so that we can use a different alt value for each image in the carousel, if it's invalid or not found). Or, another possible behavior would be to always use the first value of the array, but then why would we even accept an array as a valid value in the first place?

This opens up another discussion: should we even support an array of string as a valid type of the `imageAlt` value?

Going even further, should we even support the `imageAltField` prop at all on the component, or is this overengineered? It seems to me like a good hardcoded alt based on a field that's always going to be present (e.g., `ec_name`) would be good enough for most, if not all cases...

Reply:
In the case where the images don't work, the carousel won't work. I would only show an invalid image. So I could do one of those things : 

- Don't account for the case of an array of strings 
```
private isImageAltFieldValid() {
  return !!this.imageAlt && !Array.isArray(this.imageAlt);
}
```

- Add another condition, if it is an array. then use the first one ([0])

What do you think ?

Reply:
> In the case where the images don't work, the carousel won't work.

There's also a case to consider where some images work and some don't.

If we accept an array of strings for the alt, then IMO the "good" behavior is to require the array to have the same length as the number of images, and map each alt to the corresponding image.

If we're going to always use the first value when the alt field's value is an array, then I think it would be better to consider array of string as an invalid type for this purpose.

So I think there are three options, which I'm listing in order of personal preference:

1. We only allow string type for the `imageAltField` value.
2. We drop support the `imageAltField` prop entirely 😈 
3. We allow `string` or `string[]` for the `imageAltField` value, but if it's a `string[]`, then the array must have the same length as the number of images for the product in order for us to do the mapping, or else it will be considered invalid.
 
I feel like 1 is a good compromise, although option 2 is very tempting.

I don't like option 3 much. I think it's overengineered, but I'm listing it here because I also believe it's the only "correct" way to support the `string[]` type for the `imageAlField` value

Reply:
> I’m unclear as to why we’d choose the first alt text when an array is acceptable.

The reason is this :  ` if (this.images.length === 0) {`, if there is only 1 image, there is only 1 alt to use.

Reply:
Okay let's go with option 1 then.